internal/venice-common/src/main/java/com/linkedin/venice/controllerapi/D2ControllerClientFactory.java

```diff
@@ -51,18 +38,6 @@ public static D2ControllerClient discoverAndConstructControllerClient(
     return getControllerClient(d2ServiceName, discoveryResponse.getCluster(), d2Client);
   }
 
-  public static D2ControllerClient discoverAndConstructControllerClient(
```

Comment:
This is a breaking change, do we plan to introduce some utility method elsewhere to provide this capability? There are services that relies on this already. We should probably deprecate, move those services to the alternative implementation and then delete. Otherwise rollout of this OSS version is going to be troublesome.